### ISSUE TITLE
OpenSet Detection Issue after COCO Fine-Tuning

### ISSUE THREAD

**Issue by mio410** [AILab-CVC/YOLO-World#78]
After fine-tuning on the COCO dataset with 80 classes, when I tested the model again, I found that it could only detect the classes present in the COCO dataset. For classes outside the COCO dataset, the model failed to detect them. It seems like the model has become a closed-set detection model. Why is this happening, and is there a solution to improve the model's confidence in detecting certain classes without compromising its open-set detection capability?

**Comment by taofuyu** [comment]
I meet the same issue

**Comment by Hudaodao99** [comment]
> After fine-tuning on the COCO dataset with 80 classes, when I tested the model again, I found that it could only detect the classes present in the COCO dataset. For classes outside the COCO dataset, the model failed to detect them. It seems like the model has become a closed-set detection model. Why is this happening, and is there a solution to improve the model's confidence in detecting certain classes without compromising its open-set detection capability?

Did you get the same result as on github(YOLO-world-L map = 53.3)?

**Comment by mio410** [comment]
> > After fine-tuning on the COCO dataset with 80 classes, when I tested the model again, I found that it could only detect the classes present in the COCO dataset. For classes outside the COCO dataset, the model failed to detect them. It seems like the model has become a closed-set detection model. Why is this happening, and is there a solution to improve the model's confidence in detecting certain classes without compromising its open-set detection capability?
> 
> Did you get the same result as on github(YOLO-world-L map = 53.3)?

<img width="998" alt="截屏2024-02-27 11 08 55" src="https://github.com/AILab-CVC/YOLO-World/assets/55836251/eaf1a400-7ae1-4ae3-b8f3-3578fc5d8c7f">
This is the verification result of my last epoch.

**Comment by Hudaodao99** [comment]
> > > After fine-tuning on the COCO dataset with 80 classes, when I tested the model again, I found that it could only detect the classes present in the COCO dataset. For classes outside the COCO dataset, the model failed to detect them. It seems like the model has become a closed-set detection model. Why is this happening, and is there a solution to improve the model's confidence in detecting certain classes without compromising its open-set detection capability?
> > 
> > 
> > Did you get the same result as on github(YOLO-world-L map = 53.3)?
> 
> <img alt="截屏2024-02-27 11 08 55" width="998" src="https://private-user-images.githubusercontent.com/55836251/308023341-eaf1a400-7ae1-4ae3-b8f3-3578fc5d8c7f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkwMjc4ODUsIm5iZiI6MTcwOTAyNzU4NSwicGF0aCI6Ii81NTgzNjI1MS8zMDgwMjMzNDEtZWFmMWE0MDAtN2FlMS00YWUzLWI4ZjMtMzU3OGZjNWQ4YzdmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjI3VDA5NTMwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIyMTE5YmVkMTYwODc5MGYyYmY1YjYxNGFjYzI5NDY1MTFkNjM5YTMzYzUzYjAzMWI2ZjhlMWQ1OTVhZDIzYjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.MvsfjFlrDlHsVN8Bf5ByXFR0liNZtpIMthdmd_Q5zEM"> This is the verification result of my last epoch.

Thanks! Did u modify the config file or pth?

**Comment by HGao-cv** [comment]
> > > > After fine-tuning on the COCO dataset with 80 classes, when I tested the model again, I found that it could only detect the classes present in the COCO dataset. For classes outside the COCO dataset, the model failed to detect them. It seems like the model has become a closed-set detection model. Why is this happening, and is there a solution to improve the model's confidence in detecting certain classes without compromising its open-set detection capability?
> > > 
> > > 
> > > Did you get the same result as on github(YOLO-world-L map = 53.3)?
> > 
> > 
> > <img alt="截屏2024-02-27 11 08 55" width="998" src="https://private-user-images.githubusercontent.com/55836251/308023341-eaf1a400-7ae1-4ae3-b8f3-3578fc5d8c7f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkwMjc4ODUsIm5iZiI6MTcwOTAyNzU4NSwicGF0aCI6Ii81NTgzNjI1MS8zMDgwMjMzNDEtZWFmMWE0MDAtN2FlMS00YWUzLWI4ZjMtMzU3OGZjNWQ4YzdmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjI3VDA5NTMwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIyMTE5YmVkMTYwODc5MGYyYmY1YjYxNGFjYzI5NDY1MTFkNjM5YTMzYzUzYjAzMWI2ZjhlMWQ1OTVhZDIzYjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.MvsfjFlrDlHsVN8Bf5ByXFR0liNZtpIMthdmd_Q5zEM"> This is the verification result of my last epoch.
> 
> Thanks! Did u modify the config file or pth? And what's your bs and lr? Because I found your batch is different from mine. `02/23 21:57:06 - mmengine - INFO - Epoch(val) [80][625/625] coco/bbox_mAP: 0.4860 coco/bbox_mAP_50: 0.6560 coco/bbox_mAP_75: 0.5300 coco/bbox_mAP_s: 0.3090 coco/bbox_mAP_m: 0.5340 coco/bbox_mAP_l: 0.6400 data_time: 0.0003 time: 0.0170`

Hello, do you use mask-refine when fine-tune coco? If I don't use mask-refine, I don't get normal results. Have you ever encountered similar problems?

**Comment by LiuChuanWei** [comment]
Have you solved this problem yet?

**Comment by wondervictor** [comment]
Hi all (@mio410, @LiuChuanWei, @Hudaodao99): happy to update a milestone,
now I've tried a new setting with SGD and fewer augmentation epochs, fine-tuning without `mask-refine` or `copypaste` works.

1. reduce mosaic epochs, increase normal epochs
```python

max_epochs = 40  # Maximum training epochs
close_mosaic_epochs = 30
```

2. use SGD optimizer, add weight decay for BN and bias.
```python
optim_wrapper = dict(
    optimizer=dict(_delete_=True,
                   type='SGD',
                   lr=1e-3,
                   momentum=0.937,
                   nesterov=True,
                   weight_decay=weight_decay,
                   batch_size_per_gpu=train_batch_size_per_gpu),
    paramwise_cfg=dict(custom_keys={'logit_scale': dict(weight_decay=0.0)}),
    constructor='YOLOWv5OptimizerConstructor')
```

Under this setting, YOLO-World-Large without `mask-refine` can achieve **52.8 AP** on COCO (better than YOLOv8), and improve the former wrong baseline (48.6). BTW, fine-tuning with `mask-refine` now achieves **53.9 AP**.

This is a milestone but not the terminus and we are still working on it for a better fine-tuning setting!

Those updates will be pushed in a day.
